snippets/ram-console/README.rst

```diff
@@ -0,0 +1,37 @@
+.. _snippet-ram-console:
+
+RAM Console Snippet (ram-console)
+#################################
+
+.. code-block:: console
+
+   west build -S ram-console [...]
+
+Overview
+********
+
+This snippet redirects console output to a RAM buffer. The RAM console
+buffer is a global array lacated in RAM region by default, whose address
+is unknown before building. The RAM console driver also supports using
+a dedicated section for the RAM console buffer with prefined address.
+
+How to enable RAM console buffer section
+****************************************
+
+Add board dts overlay to this snippet to add property ``zephyr,ram-console``
+in the chosen node and memory-region node with compatible string
+:dtcompatible:`zephyr,memory-region` as the following:
+
+.. code-block:: DTS
+
+    / {
+        chosen {
+            zephyr,ram-console = &snippet_ram_console;
+        };
+
```

Comment:
Instead of repeating this in each board overlay, it would probably make sense to move it to a common `ram-console.overlay`

Reply:
By default, the RAM console uses a global array for the console buffer, it doesn't need this chosen property. It is only needed when the RAM console wants to use a dedicated section for console buffer.